src/index.js

```diff
@@ -0,0 +1,159 @@
+"use strict";
+
+//to keep track of and to be able to modify current state
+const state = {
+    city: 'SanFrancisco',
+    temp: 60,
+    latitude: null,
+    longitude: null
+}
+
+// would love to not need this and change the units in params instead
+const convertKtoF = (temp) => {
+    return (temp - 273.15) * (9 / 5) + 32;
+  };
+
+// function to get input city's lat and long
+const getLocation = () => {
+    // location = state.city
+    // from axios documentation, the get request has a PATH and then otional params, more info: https://github.com/axios/axios
+    axios.get('http://localhost:5000/location', {
+        params: {q: state.city}
+    }) 
+    .then(response => {
+    // per axios documentation:`data` is the response that was provided by the server \\data: {}
+    //store current lon and lat in state
+    state.latitude = response.data[0].lat;
+    state.longitude = response.data[0].lon;
+    getWeather();
+    })
+    .catch(error =>{
+        console.log("Location error: ", error.response.data)
+    });
+
+}
+
+// function to get input city's weather based on its lat and long - called from getLocation
+const getWeather = () => {
+    // latitude, longitude = current state
+    axios.get('http://localhost:5000/weather', {
+        params: {lat: state.latitude, lon: state.longitude}
+    }) 
+    .then(response => {
+    // check out how a response looks: https://openweathermap.org/current#geo or the replit I shared
+    const weather = response.data.main.temp
+    // couldn't get adding a unit (fahrenheit) param to work for temperature, there is an option in the docs
+    state.temp = Math.round(convertKtoF(weather));
+    changeTempColorAndLandscape();
+    })
+    .catch(error =>{
+        console.log("Weather error: ", error.response.data)
+    });
+
+}
+
+//functions to increase and decrease the temp
+const increaseTemp = () => {
+    state.temp += 1;
+    changeTempColorAndLandscape()
+  }
+const decreaseTemp = () => {
+    state.temp -= 1;
+    changeTempColorAndLandscape()
+  }
+
+//function to change sky
+const changeSkyDisplay = () => {
+    //getting selected sky image
+  const input = document.getElementById('sky_select').value;
+// selecting the sky_image element object
+  const currentSky = document.getElementById('sky_image');
+  let skyDisplay =	'🌧🌈⛈🌧🌧💧⛈🌧🌦🌧💧🌧🌧';
+
+  if (input === 'Sunny'){
+    skyDisplay =	'☁️ ☁️ ☁️ ☀️ ☁️ ☁️';
+  }else if (input === 'Cloudy'){
+    skyDisplay =	'☁️☁️ ☁️ ☁️☁️ ☁️ 🌤 ☁️ ☁️☁️';
+  }else if (input === 'Rainy'){
+    skyDisplay =	'🌧🌈⛈🌧🌧💧⛈🌧🌦🌧💧🌧🌧';
+  }else if (input === 'Snowy'){
+    skyDisplay =	'🌨❄️🌨🌨❄️❄️🌨❄️🌨❄️❄️🌨🌨';
+  };
```

Comment:
How might you refactor this so the logic is more data driven? Here you have several if/else if statements. What if you had an object with the keys as the type of weather and the values as the corresponding display? Then you could iterate over the object to set `skyDisplay`